plugins/scaffolder/src/components/fields/EntityTagsPicker/EntityTagsPicker.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import { useAsync, useEffectOnce } from 'react-use';
+import { CatalogEntitiesRequest } from '@backstage/catalog-client';
+import { Entity, makeValidator } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { FormControl, TextField } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import { FieldProps } from '@rjsf/core';
+
+export const EntityTagsPicker = ({
+  formData,
+  onChange,
+  uiSchema,
+}: FieldProps<string[]>) => {
+  const catalogApi = useApi(catalogApiRef);
+  const [inputValue, setInputValue] = useState('');
+  const [inputError, setInputError] = useState(false);
+  const tagValidator = makeValidator().isValidTag;
```

Comment:
This is a bit tricky since it's possible to override this in the catalog. I think we could keep as is and see if someone runs into an issue with it, but just highlighting that catalog validation on the client perhaps isn't a completely solved problem x)

@backstage/maintainers thinking stuff like a validation endpoint, or a `catalogValidationApiRef`

Reply:
Yeah we will probably sit down internally and talk more about this same thing soon right? It's worth some consideration. The api ref is a neat idea to start with since it can be made without having to invent a backend supporting it right away